public/app.js

```diff
@@ -0,0 +1,380 @@
+const APIController = (function() {
+
+  const clientId = 'e91988fcc1674aa88374f24be4844f85';
+  const clientSecret = '10badbb697f544ecb15e1efd7659a0b0';
+
+  const send = async (url, token, category,params) => {
+    if (!params) {
+      params = {
+        method: 'GET',
+        headers: {
+          'Authorization': 'Bearer ' + token
+        }
+      };
+    }
+    try {
+      const result = await fetch(`${url}`, params);
+
+      const data = await result.json();
+
+      if (data.error) {
+         alert("We cant load info,'couse of Spotify's troubles");
+        return null;
+      } else
+        switch (category) {
+          case 'Genre': return data.categories.items;
+          case 'GenrePlaylists': return data.playlists.items;
+          case 'Songs': return data.items;
+          default: return data;
+
+        }
+    } catch (error) {
+      alert(error.message);
+    }
+    return null;
+  }
+  /**
+   * @description Функция получения Токена Spotify
+   */
+  const _getToken = async () => {
+
+    const result = await fetch('https://accounts.spotify.com/api/token', {
```

Comment:
Сейчас есть метод send

Сейчас с помощью него все можно сделать
```suggestion
    const result = send('https://accounts.spotify.com/api/token', null, null, {
       method: 'POST',
       headers: {
         'Content-Type': 'application/x-www-form-urlencoded',
         'Authorization': 'Basic ' + btoa(clientId + ':' + clientSecret)
       },
       body: 'grant_type=client_credentials'
     })
     if (result) {
         return result.access_token;
     }
     return null;
```